src/charm.py

```diff
@@ -258,11 +261,15 @@ def _update_workload_container_config(self, event):
             },
         }
         layer_label = "livepatch"
+        force_restart = self._update_trusted_ca_certs(workload_container)
         workload_container.add_layer(layer_label, update_config_environment_layer, combine=True)
         if self._ready(workload_container):
             if workload_container.get_service(LIVEPATCH_SERVICE_NAME).is_running():
-                LOGGER.info("Replanning services")
-                workload_container.replan()
+                if force_restart:
+                    workload_container.restart(LIVEPATCH_SERVICE_NAME)
+                else:
+                    LOGGER.info("Replanning services")
+                    workload_container.replan()
```

Comment:
The problem is this `_update_workload_container_config` method already has many branches (12 allowed by pylint). So, adding this if-statements results in the lint issue. We should either move some blocks into separate methods, or try merge/remove some branches.

Reply:
yeah.. simplified that method a bit.. it's way too complicated